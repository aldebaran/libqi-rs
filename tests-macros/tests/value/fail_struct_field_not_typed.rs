struct F;

#[derive(
    qi_macros::AsValue,
    qi_macros::FromValue,
    qi_macros::IntoValue,
    qi_macros::Reflect,
    qi_macros::StdTryFromValue,
)]
struct S {
    pub f: F,
}

fn main() {}
