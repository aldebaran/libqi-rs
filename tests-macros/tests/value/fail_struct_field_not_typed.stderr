error[E0277]: the trait bound `F: AsValue` is not satisfied
 --> tests/value/fail_struct_field_not_typed.rs:4:5
  |
4 |     qi_macros::AsValue,
  |     ^^^^^^^^^^^^^^^^^^ the trait `AsValue` is not implemented for `F`
  |
  = help: the following other types implement trait `AsValue`:
            &T
            &mut T
            ()
            (T0, T1)
            (T0, T1, T2)
            (T0, T1, T2, T3)
            (T0, T1, T2, T3, T4)
            (T0, T1, T2, T3, T4, T5)
          and $N others
  = note: this error originates in the derive macro `qi_macros::AsValue` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `F: FromValue<'_>` is not satisfied
 --> tests/value/fail_struct_field_not_typed.rs:5:5
  |
5 |     qi_macros::FromValue,
  |     ^^^^^^^^^^^^^^^^^^^^ the trait `FromValue<'_>` is not implemented for `F`
  |
  = help: the following other types implement trait `FromValue<'a>`:
            <&'a [u8] as FromValue<'a>>
            <&'a str as FromValue<'a>>
            <() as FromValue<'_>>
            <(T0, T1) as FromValue<'a>>
            <(T0, T1, T2) as FromValue<'a>>
            <(T0, T1, T2, T3) as FromValue<'a>>
            <(T0, T1, T2, T3, T4) as FromValue<'a>>
            <(T0, T1, T2, T3, T4, T5) as FromValue<'a>>
          and $N others
  = note: this error originates in the derive macro `qi_macros::FromValue` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Value<'_>: From<F>` is not satisfied
 --> tests/value/fail_struct_field_not_typed.rs:6:5
  |
6 |     qi_macros::IntoValue,
  |     ^^^^^^^^^^^^^^^^^^^^ the trait `From<F>` is not implemented for `Value<'_>`
  |
  = help: the following other types implement trait `From<T>`:
            <Value<'_> as From<()>>
            <Value<'_> as From<PhantomData<T>>>
            <Value<'_> as From<Signature>>
            <Value<'_> as From<[T; 0]>>
            <Value<'_> as From<bool>>
            <Value<'_> as From<bytes::bytes::Bytes>>
            <Value<'_> as From<f32>>
            <Value<'_> as From<f64>>
          and $N others
  = note: this error originates in the derive macro `qi_macros::IntoValue` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `F: Reflect` is not satisfied
  --> tests/value/fail_struct_field_not_typed.rs:11:12
   |
11 |     pub f: F,
   |            ^ the trait `Reflect` is not implemented for `F`
   |
   = help: the following other types implement trait `Reflect`:
             &str
             ()
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
             (T0, T1, T2, T3, T4)
             (T0, T1, T2, T3, T4, T5)
             (T0, T1, T2, T3, T4, T5, T6)
           and $N others
